src/packages/utils/src/things/json-rpc/json-rpc-quantity.ts

```diff
@@ -91,59 +113,153 @@ export class Quantity extends BaseJsonRpcType {
     if (length > 6) {
       const trimmedBuffer =
         firstNonZeroByte === 0
-          ? this.bufferValue
-          : this.bufferValue.subarray(firstNonZeroByte, length);
+          ? bufferValue
+          : bufferValue.subarray(firstNonZeroByte);
+
       result = Number(bufferToBigInt(trimmedBuffer));
 
       if (!Number.isSafeInteger(result)) {
         console.warn(
-          `0x${this.bufferValue.toString(
+          `0x${bufferValue.toString(
             "hex"
           )} is too large - the maximum safe integer value is 0${Number.MAX_SAFE_INTEGER.toString(
             16
           )}`
         );
       }
     } else {
-      result = this.bufferValue.readUIntBE(firstNonZeroByte, length);
+      result = bufferValue.readUIntBE(firstNonZeroByte, length);
     }
 
     return result;
   }
 
   valueOf(): bigint {
-    if (this.bufferValue == null) {
+    if (this._lazyBufferValue.getValue() == null) {
       return null;
     } else {
       return this.toBigInt();
     }
   }
 
   private findFirstNonZeroByteIndex(): number {
+    const bufferValue = this._lazyBufferValue.getValue();
     let firstNonZeroByte = 0;
     for (
       firstNonZeroByte = 0;
-      firstNonZeroByte < this.bufferValue.length;
+      firstNonZeroByte < bufferValue.length;
       firstNonZeroByte++
     ) {
-      if (this.bufferValue[firstNonZeroByte] !== 0) break;
+      if (bufferValue[firstNonZeroByte] !== 0) break;
     }
     return firstNonZeroByte;
   }
 
-  static toBuffer(value: JsonRpcInputArg, nullable?: boolean): Buffer {
+  private static getArgumentAsBigInt(argument: JsonRpcInputArg | Quantity): bigint {
+    const argumentType = typeof argument;
+    let argumentValue: bigint;
+
+    switch(argumentType) {
```

Comment:
Do this:

```suggestion
    let argumentValue: bigint;

    switch (typeof argument) {
```

and then remove the `as number`s below.